### PR TITLE
Remove Legacy code in ubipop  [RHELDST-13337]

### PR DESCRIPTION
**Pull request by jihupdc**
1.Legacy code in ubipop is removed.
2.ubi-manifest is the only source of expected content for ubipop.

### CODE REVIEW

File: ubipop/cli.py [hunk 1]
```diff
@@ -112,14 +112,10 @@ def parse_args(args):
...
 
-FOREIGN_LOGGERS = ("pubtools.pulplib", "fastpurge", "ubiconfig")
-
-
 def main(args):
     logging.basicConfig(format=DEFAULT_LOG_FMT, datefmt=DEFAULT_DATE_FMT)
-    for logger in FOREIGN_LOGGERS:
-        logging.getLogger(logger).setLevel(logging.INFO)
-
+    logging.getLogger("pubtools.pulplib").setLevel(logging.INFO)
```
Comment (rbikar) [inline ubipop/cli.py:117]: Why is this in diff? You shouldn't touch that.
Please rebase on master

File: ubipop/__init__.py [hunk 2]
```diff
@@ -421,10 +412,7 @@ def _get_ubi_repo_sets(self, ubi_binary_cs):
         return ubi_repo_sets
 
     def _get_population_sources(self, out_repo):
-        if (
-            not hasattr(out_repo, "population_sources")
-            or not out_repo.population_sources
-        ):
+        if not out_repo.population_sources:
```
Comment (rbikar) [inline ubipop/__init__.py:415]: This is also unrelated. Please rebase on master.

File: .pylintrc [hunk 3]
```diff
@@ -102,5 +102,12 @@ disable=print-statement,
         # skip check for non-snake_case names
         invalid-name,
         # skip check for lambdas
-        unnecessary-lambda
+        unnecessary-lambda,
```
Comment (rbikar) [inline .pylintrc:105]: Any comment to this?

**Comment by rbikar** [comment]
Please check coverage report whether the reported problems are new or if they were not covered previously as well.
Also rebase on current on master, you are currently reverting some change already merged in master.

**Review by rbikar — CHANGES_REQUESTED** [review]

File: ubipop/__init__.py (hunk 2 above)
Comment (jihupdc) [reply ubipop/__init__.py]: Done

**Comment by rbikar** [comment]
Please see detailed codecov report https://app.codecov.io/gh/release-engineering/ubi-population-tool/pull/199 and please fix parts of code that are newly not covered by tests.

File: ubipop/__init__.py [hunk 4]
```diff
@@ -98,8 +97,7 @@ def _ensure_repos_existence(self):
             _LOG.error("Output Source repo does not exist")
 
         if not self.out_repos.debug:
-            fatal = True
-            _LOG.error("Output Debug repo does not exist")
+            _LOG.warning("Output Debug repo does not exist")
```
Comment (rbikar) [inline ubipop/__init__.py:100]: This seems like you are reverting some changes from master, please check.

**Review by rbikar — CHANGES_REQUESTED** [review]

**Comment by rbikar** [comment]
@jihupdc 
Please rebase on master so the py27 tests can pass.

File: ubipop/__init__.py (hunk 4 above)
Comment (jihupdc) [reply ubipop/__init__.py]: Done

File: ubipop/__init__.py [hunk 5]
```diff
@@ -86,8 +86,7 @@ def _ensure_repos_existence(self):
             _LOG.error("Input Source repo does not exist")
 
         if not self.in_repos.debug:
-            fatal = True
-            _LOG.error("Input Debug repo does not exist")
+            _LOG.warning("Input Debug repo does not exist")
```
Comment (rbikar) [inline ubipop/__init__.py:89]: This also shouldn't be part of this change.

**Review by rbikar — CHANGES_REQUESTED** [review]

File: ubipop/__init__.py (hunk 5 above)
Comment (jihupdc) [reply ubipop/__init__.py]: Done

**Comment by rbikar** [comment]
@jihupdc 
I think this PR needs another rebase on master to fix the failing test or maybe another fix (ref. https://github.com/release-engineering/ubi-population-tool/pull/198/files )

File: tests/test_ubipop.py [hunk 6]
```diff
@@ -347,40 +346,6 @@ def test_get_ubi_repo_sets(get_debug_repository, get_source_repository):
     assert output_repos.debug.id == "ubi_debug"
 
 
-@patch("pubtools.pulplib.YumRepository.get_debug_repository")
-@patch("pubtools.pulplib.YumRepository.get_source_repository")
-def test_get_ubi_repo_sets_repo_missing(
```
Comment (rbikar) [inline tests/test_ubipop.py:352]: What's the reason for removing this test case?

**Review by rbikar — CHANGES_REQUESTED** [review]
I believe the removed test case is still relevant and should be kept in place.

**Comment by rbikar** [comment]
run tests

**Review by rbikar — APPROVED** [review]

**Comment by rbikar** [comment]
jenkins CI seems to be still broken

**Pull request merged**